warehouse/models/intermediate/gtfs/gtfs_quality/int_gtfs_quality__shapes_file_present.sql

```diff
@@ -0,0 +1,32 @@
+WITH feed_guideline_index AS (
+    SELECT * FROM {{ ref('int_gtfs_quality__schedule_feed_guideline_index') }}
+),
+
+keyed_parse_outcomes AS (
+    SELECT * FROM {{ ref('int_gtfs_schedule__keyed_parse_outcomes')}}
+),
+
+daily_feed_shapes_present AS (
+    SELECT feed_key
+      FROM keyed_parse_outcomes
+     WHERE parse_success
+       AND filename = 'shapes.txt'
+       AND feed_key IS NOT null
+),
+
+int_gtfs_quality__shapes_file_present AS (
+    SELECT
+        idx.date,
+        idx.feed_key,
+        {{ shapes_file_present() }} AS check,
+        {{ accurate_service_data() }} AS feature,
+        CASE
+            WHEN s.feed_key IS NOT null THEN "PASS"
+            ELSE "FAIL"
```

Comment:
Not sure if you want to leave a null option for feeds that failed to download? (Would need to go back to joining in the overall feed download outcome to check for "feed present but doesn't have shapes" vs. "feed not present").

Reply:
In retrospect I may rescind this proposal. I think need two grains:

* Checks assessed against feeds that actually downloaded
* Checks assessed against URLs where download was attempted (this is basically just the "did it download" check) 

But I think maybe we want to just assume that the universe for any check that depends on a feed existing is "feeds that exist" and we can handle the non-existent case in a later join where we assemble the larger universe. If that makes sense

Reply:
I don't entirely understand but I'm happy to follow your advice for now. It might make sense for us to meet and discuss, especially since so many questions are coming up in migration.